### PR TITLE
Close streams when Managed* classes are deleted

### PR DESCRIPTION
**Pull request by Frassle**
Add "leaveOpen" parameter to opt out of this (same parameter name as
StreamWriter/Reader).

### CODE REVIEW

File: cpp/ManagedOutputStream.cpp [hunk 1]
```diff
@@ -34,6 +35,10 @@ class ManagedOutputStream final : public arrow::io::OutputStream
 
 	~ManagedOutputStream()
 	{
+		arrow::Status st = this->Close();
+		if (!st.ok()) {
+			ARROW_LOG(FATAL) << "Error ignored when destroying ManagedOutputStream: " << st;
```
Comment (herebebeasties) [inline cpp/ManagedOutputStream.cpp]: Should this be FATAL if it's ignored and we carry on regardless? ERROR perhaps?

Comment (Frassle) [reply cpp/ManagedOutputStream.cpp]: Yes it should be ERROR.

**Comment by GPSnoopy** [comment]
I'll have some time to work on ParquetSharp end of next week. We've got a PR update to vcpkg Arrow that has been accepted, the latest releases of Arrow fixes quite a few Parquet bugs.

**Comment by GPSnoopy** [comment]
Being merged into master as part of PR #77. Nuget package already built: https://www.nuget.org/packages/ParquetSharp/2.0.2-beta2

**Pull request merged**